mrp_subcontracting_skip_no_negative/models/stock_move.py

```diff
@@ -14,11 +14,21 @@ def _action_done(self, cancel_backorder=False):
         # For rather unlikely occassions where linked production is not in the right
         # state.
         for move in moves_with_no_check:
-            production_move = self.search([("move_dest_ids", "=", move.id)])
-            production = production_move.production_id
-            if production.reservation_state != "assigned":
+            production_move = self.search(
+                [
+                    ("move_dest_ids", "=", move.id),
+                    ("state", "not in", ("done", "cancel")),
+                ]
+            )
+            productions = production_move.production_id
+            unassigned_productions = productions.filtered(
+                lambda p: p.reservation_state != "assigned"
+            )
+            for production in unassigned_productions:
                 production.action_assign()
-            if production.reservation_state == "assigned":
+            if all(
+                state == "assigned" for state in productions.mapped("reservation_state")
```

Comment:
I guess this is a trivial point and may not matter.
```suggestion
                state == "assigned" for state in unassigned_productions.mapped("reservation_state")
```

Reply:
it looks much better like this, thanks!